pkg/ccl/backupccl/restore_job.go

```diff
@@ -221,6 +223,65 @@ func makeBackupLocalityMap(
 	return backupLocalityMap, nil
 }
 
+func filterCompletedImportSpans(
+	completedSpansSlice []jobspb.RestoreProgress_RestoreProgressFrontierEntry,
+	importSpans []execinfrapb.RestoreSpanEntry,
+) (*spanUtils.Frontier, []execinfrapb.RestoreSpanEntry, error) {
+	var frontierCreationSlice []roachpb.Span
+	var modifiedImportSpans []execinfrapb.RestoreSpanEntry
+	for _, frontierSpanEntry := range completedSpansSlice {
+		frontierCreationSlice = append(frontierCreationSlice, frontierSpanEntry.Entry)
+	}
+	checkpointingSpanFrontier, err := spanUtils.MakeFrontier(frontierCreationSlice...)
+	if err != nil {
+		return checkpointingSpanFrontier, modifiedImportSpans, err
```

Comment:
nit: given that the restore will fail immediately if this func fails, modify the return signatures during an error to:
```
return nil, nil, err
```